src/measure.rs

```diff
@@ -116,92 +162,165 @@ async fn create_provider(endpoint: &Endpoint, api: &Api) -> Result<Provider> {
-    let mut provider =
-        create_provider(&measurement_config.endpoint, &measurement_config.api).await?;
-
-    let signals = provider
-        .provider_interface
-        .as_mut()
-        .validate_signals_metadata(&measurement_config.config_signals)
-        .await
-        .unwrap();
-
-    let subscriber = setup_subscriber(
-        &measurement_config.endpoint,
-        signals.clone(),
-        &measurement_config.api,
-    )
-    .await?;
-
-    let hist = Histogram::<u64>::new_with_bounds(1, 60 * 60 * 1000 * 1000, 3)?;
-    let running_hist = Histogram::<u64>::new_with_bounds(1, 60 * 60 * 1000 * 1000, 3)?;
-
-    let start_time = SystemTime::now();
-
-    let progress = if measurement_config.run_forever {
-        ProgressBar::new_spinner().with_style(
-            // TODO: Add average latency etc...
-            ProgressStyle::with_template("[{elapsed_precise}] {wide_msg} {pos:>7} iterations")
-                .unwrap(),
+    let shutdown_handler_ref = Arc::new(RwLock::new(shutdown_handler));
+
+    let mut tasks: JoinSet<Result<MeasurementResult>> = JoinSet::new();
+
+    let multi_progress_bar = MultiProgress::new();
+
+    for group in config_groups.clone() {
+        let shutdown_handler = Arc::clone(&shutdown_handler_ref);
+
+        let (initial_values_sender, mut initial_values_reciever) =
+            tokio::sync::mpsc::channel::<HashMap<String, DataValue>>(10);
+
+        let provider_endpoint =
+            create_databroker_endpoint(measurement_config.host.clone(), measurement_config.port)?;
+        let subscriber_endpoint =
+            create_databroker_endpoint(measurement_config.host.clone(), measurement_config.port)?;
+        let mut measurement_config = measurement_config.clone();
+
+        measurement_config.interval = group.cycle_time_ms;
+        let mut provider = create_provider(&provider_endpoint, &measurement_config.api).await?;
+
+        let signals = provider
+            .provider_interface
+            .as_mut()
+            .validate_signals_metadata(&group.signals)
+            .await
+            .unwrap();
+
+        let subscriber = setup_subscriber(
+            &subscriber_endpoint,
+            signals.clone(),
+            &measurement_config.api,
+            initial_values_sender,
         )
-    } else {
-        ProgressBar::new(measurement_config.iterations).with_style(
-            ProgressStyle::with_template(
-                "[{elapsed_precise}] {msg} [{wide_bar}] {pos:>7}/{len:7} iterations",
+        .await?;
+
+        // Receive the initial signal values
+        if let Some(initial_signal_values) = initial_values_reciever.recv().await {
+            let result = provider
+                .provider_interface
+                .as_mut()
+                .set_initial_signals_values(initial_signal_values)
+                .await;
+            assert!(result.is_ok());
+        }
+
+        let hist = Histogram::<u64>::new_with_bounds(1, 60 * 60 * 1000 * 1000, 3)?;
```

Comment:
Possibly give a comment what "60*60*1000*1000" means.

Reply:
@argerus can you help with that?

Reply:
After looking a bit at the code I assume this means that we record values between 1 microsecond and 1 hour, with 3 significant digits (when represented as microseconds)